File: mobile/android/app/src/main/java/com/pedidoprime/centraloperacional/data/MockOperationalRepository.kt
package com.pedidoprime.centraloperacional.data

import com.pedidoprime.centraloperacional.model.ModuleId
import com.pedidoprime.centraloperacional.model.OperationalMetrics
import com.pedidoprime.centraloperacional.model.OperationalModule
import com.pedidoprime.centraloperacional.model.OperationalSnapshot
import com.pedidoprime.centraloperacional.model.OperationalUser
import kotlin.random.Random
import kotlinx.coroutines.delay

/**
 * Repositório sem backend: simula atualizações em tempo real mutando os
 * números aleatoriamente a cada [fetchSnapshot], com uma latência simulada
 * de rede — permite rodar e demonstrar o app inteiro sem nenhuma API.
 */
class MockOperationalRepository(
    override val refreshIntervalMs: Long = 5000
) : OperationalRepository {

    private var current = seedSnapshot

    override suspend fun fetchSnapshot(): OperationalSnapshot {
        delay(350)
        current = current.mutatedRandomly()
        return current
    }

    companion object {
        private val seedSnapshot = OperationalSnapshot(
            user = OperationalUser.PLACEHOLDER,
            metrics = OperationalMetrics(mesasAbertas = 8, emPreparo = 5, turnoAtual = 1847.0),
            modules = listOf(
                OperationalModule(ModuleId.PEDIDOS, count = 12),
                OperationalModule(ModuleId.COZINHA, count = 5),
                OperationalModule(ModuleId.BAR, count = 3),
                OperationalModule(ModuleId.CAIXA, count = 0)
            )
        )
    }
}

private fun Int.walkedRandomly(range: IntRange = -1..2): Int =
    (this + Random.nextInt(range.first, range.last + 1)).coerceAtLeast(0)

private fun OperationalMetrics.mutatedRandomly(): OperationalMetrics = copy(
    mesasAbertas = mesasAbertas.walkedRandomly(),
    emPreparo = emPreparo.walkedRandomly(),
    turnoAtual = (turnoAtual + Random.nextInt(-20, 45)).coerceAtLeast(0.0)
)

private fun OperationalModule.mutatedRandomly(): OperationalModule = copy(count = count.walkedRandomly())

private fun OperationalSnapshot.mutatedRandomly(): OperationalSnapshot = copy(
    metrics = metrics.mutatedRandomly(),
    modules = modules.map { it.mutatedRandomly() }
)

File: mobile/ios/CentralOperacionalApp/Services/MockOperationalRepository.swift
import Foundation

/// Implementação sem backend — permite rodar e demonstrar o app sozinho.
/// Varia levemente os números a cada busca para simular atividade real.
final class MockOperationalRepository: OperationalRepository {
    let refreshInterval: TimeInterval

    private var current: OperationalSnapshot

    init(refreshInterval: TimeInterval = 5, seed: OperationalSnapshot = .mock) {
        self.refreshInterval = refreshInterval
        self.current = seed
    }

    func fetchSnapshot() async throws -> OperationalSnapshot {
        try await Task.sleep(nanoseconds: 350_000_000) // simula latência de rede
        current.metrics = current.metrics.mutatedRandomly()
        current.modules = current.modules.map { $0.mutatedRandomly() }
        return current
    }
}

private extension OperationalMetrics {
    /// Passeio aleatório pequeno em torno do valor atual — números plausíveis,
    /// nunca negativos, dando a sensação de dado "vivo" sem backend real.
    func mutatedRandomly() -> OperationalMetrics {
        OperationalMetrics(
            mesasAbertas: max(0, mesasAbertas + Int.random(in: -1...1)),
            emPreparo: max(0, emPreparo + Int.random(in: -1...1)),
            turnoAtual: turnoAtual + Decimal(Double.random(in: 0...45))
        )
    }
}

private extension OperationalModule {
    func mutatedRandomly() -> OperationalModule {
        OperationalModule(id: id, count: max(0, count + Int.random(in: -1...1)))
    }
}

extension OperationalSnapshot {
    static let mock = OperationalSnapshot(
        user: .placeholder,
        metrics: OperationalMetrics(mesasAbertas: 8, emPreparo: 5, turnoAtual: 1847),
        modules: [
            OperationalModule(id: .pedidos, count: 12),
            OperationalModule(id: .cozinha, count: 5),
            OperationalModule(id: .bar, count: 3),
            OperationalModule(id: .caixa, count: 0),
        ]
    )
}
